### DISCUSSION TITLE
Better server GUI

### DISCUSSION THREAD

**Discussion by YouHaveTrouble** [PurpurMC/Purpur#478] [Feature Requests]
As discussed in discord, the idea came up to swap default server gui to a website based one.

This would include all default GUI's functionalities and would expand them further.

While we're thinking of hosting a website for a server panel like that, there is also a possibility to include REST api that would provide server information to use on server websites or apps easily.

The discord discussion that started the idea: [here](https://discord.com/channels/685683385313919172/685683802479001625/861991062775136274)

**Comment by katekerllenevich** [comment]
here are 2 patches from the 1.15 era

https://github.com/pl3xgaming/Purpur/blob/ver/1.15/patches/server/0044-Make-the-GUI-better.patch
https://github.com/pl3xgaming/Purpur/blob/ver/1.15/patches/server/0088-UPnP-Port-Forwarding-Service.patch
